### ISSUE TITLE
useParams and other new hooks are not exported in react-router-dom & react-router

### ISSUE THREAD

**Issue by ironbyte** [ReactTraining/react-router#6944]
## Version
    
"react-router-dom": "^5.1.0",
"react-scripts": "3.1.1", (CRA)

## Steps to reproduce

Copied from https://reacttraining.com/blog/react-router-v5-1/:

```
import { BrowserRouter as Router, Route, Switch, useParams } from 'react-router-dom'

function BlogPost() {
  let { slug } = useParams();
  // ...
}

ReactDOM.render(
  <Router>
    <div>
      <Switch>
        {/* No weird props here, just use
            regular `children` elements! */}
        <Route path="/posts/:slug">
          <BlogPost />
        </Route>
      </Switch>
    </div>
  </Router>,
  document.getElementById('root')
)
```

## Expected Behavior

The slug variable should be initialized with the actual slug ID from the slug ID parameter of a given URL.

## Actual Behavior

Error:

Attempted import error: 'useParams' is not exported from 'react-router-dom'.

**Comment by MeiKatz** [comment]
1st – where in your example do you use `useParams`?
2nd – if you want to use the props of a `<Route>` inside a component you currently have to use the `component` prop. `children` don't get them passed. @mjackson: I think we should fix this so we can drop `render` and `component` in RR 6.

**Comment by ironbyte** [comment]
Hey, I've corrected the codes in my example. I'm happy to report that I've fixed the issue by removing node_modules directory and yarn.lock and then running yarn to re-install dependencies. I'm now able to use the new hooks without issues.

**Comment by olalonde** [comment]
I'm having same issue

**Comment by jpascal** [comment]
+1 to issue

**Comment by pravintamil** [comment]
Hi Guys,
Actually the `useParams` is exported from `react-router` instaed of `react-router-dom`. So please use below line for `useParams`
``` javascript
import React from "react";
import { useParams } from "react-router";
import { Route, Switch, Link } from "react-router-dom";
```

**Comment by pshrmn** [comment]
The hook are re-exported by `react-router-dom`:

If you look at the [published package code](https://unpkg.com/browse/react-router-dom@5.1.2/esm/react-router-dom.js), you will see:
```js
export * from 'react-router';
```

You should always import from `react-router-dom`.

**Comment by NYCJacob** [comment]
well this is closed but I am getting the same issue even after deleting node_module and reinstalling, but fixed when I import from react-router

**Comment by Cst2989** [comment]
Just get the id from the props. 

props.match.params.id;

**Comment by tylerthome** [comment]
> Just get the id from the props.
> 
> props.match.params.id;

At least in my case, the whole reason for "useParams" is so I don't have to join RouteComponentProps onto my function component props interfaces every time.

**Comment by KKS1** [comment]
For anyone else facing this issue, **`yarn upgrade react-router-dom`** fixed this for me. 
Basically, what it does is update the `react-router` dependency version as well, so as your application is running all up-to-date versions for using the latest hooks from `react-router-dom`.

**Comment by andreas0607** [comment]
> For anyone else facing this issue, **`yarn upgrade react-router-dom`** fixed this for me.
> Basically, what it does is update the `react-router` dependency version as well, so as your application is running all up-to-date versions for using the latest hooks from `react-router-dom`.

Adding to what KKS1 is saying. I had to run `yarn upgrade react-router-dom` and `yarn upgrade @types/react-router-dom`. Though this only applies if you need type-definitions.

**Comment by mjackson** [comment]
tl;dr: Use the right version of react-router-dom. The hooks are in there.
